### DISCUSSION TITLE
Generating JWT from framework

### DISCUSSION THREAD

**Discussion by xttlegendapi** [pocketbase/pocketbase#2799] [Q&A] [answered]
Hello
Is it possible to generate a JWT for a user from the go package (using as framework)?

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the helpers from the [`tokens` package](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.6/tokens), for example:

```go
record, err := app.Dao().FindAuthRecordByEmail("users", "test@example.com")
if err != nil {
	return err
}

token, err := tokens.NewRecordAuthToken(app, record)
if err != nil {
	return err
}
```

_Alternatively you can also generate your own non-auth tokens using directly [`security.NewToken()`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.6/tools/security#NewToken). You can check the code for record tokens for examples - https://github.com/pocketbase/pocketbase/blob/master/tokens/record.go_.
